src/types/search.ts

```diff
@@ -1,5 +1,13 @@
+import { ShareEntity } from './common';
+
 export enum SearchFilters {
   None = 0,
   SharedByMe = 1 << 0,
   PublishedByMe = 1 << 1,
 }
+
+export type EntityFilter<T> = (item: T) => boolean;
+export interface EntityFilters {
+  rootFilter: EntityFilter<ShareEntity>; // filter root level folders and items e.g. "Shared with me"
```

Comment:
Do we really need this differentiation? And how it will work? Could you provide more examples?